src/librustc_codegen_llvm/llvm/ffi.rs

```diff
@@ -1358,6 +1358,7 @@ extern "C" {
     pub fn LLVMRustDebugMetadataVersion() -> u32;
     pub fn LLVMRustVersionMajor() -> u32;
     pub fn LLVMRustVersionMinor() -> u32;
+    pub fn LLVMRustIsRustLLVM() -> bool;
```

Comment:
I'm surprised we don't have this function already anywhere 🤔

Reply:
Somewhat surprisingly, there is only a single use of `LLVM_RUSTLLVM` apart from the one introduced here. I guess the need to know about this from Rust code never came up.

I could also change the code to check for LLVM 8.0 instead and save this function. I only went for the RUSTLLVM check here because this is based on a cherry-picked commit, but ultimately checking for 8.0 would achieve the same and save some churn down the line.